hs_modelinstance/models.py

```diff
@@ -58,20 +58,50 @@ def modelProgramIdentifier(self):
 
     @classmethod
     def create(cls, **kwargs):
-        shortid = kwargs['model_name']
-        # get the MP object that matches.  Returns None if nothing is found
-        obj = ModelProgramResource.objects.filter(short_id=shortid).first()
-        metadata_obj = kwargs['content_object']
-        title = obj.title
-        return super(ExecutedBy,cls).create(model_program_fk=obj, model_name=title,content_object=metadata_obj)
+
+        # when creating a new resource version, we need to lookup the existing model_program_fk
+        if 'model_program_fk' in kwargs:
+
+            # get the foreign key id if one has been set (i.e. when creating a new version)
+            mp_short_id = kwargs['model_program_fk']
+
+            # get the MP object that matches.  Returns None if nothing is found
+            obj = ModelProgramResource.objects.filter(id=mp_short_id).first()
+
+        # when adding or changing the model_program_fk, we need to lookup the model obj based on mp_short_id
+        else:
+            # get the selected model program short id that has been submitted via the form
+            mp_short_id = kwargs['model_name']
```

Comment:
This variable 'model_name' is confusing me as its name implies it should carry a name but looks like it carries a resource id. And this class has a field named "model_name" as well, which is supposed to be the name/title of a specific model program resource referenced by this ExecutedBy metadata obj, right?
If I understood it correctly, maybe it is better to rename the first "model_name" to avoid ambiguity.